aten/src/THCUNN/ClassNLLCriterion.cu

```diff
@@ -125,7 +125,7 @@ __global__ void cunn_ClassNLLCriterion_updateOutput_kernel(Dtype *output,
     }
     *total_weight = ScalarConvert<Acctype, Dtype>::to(total_weightAcc);
     *output = ScalarConvert<Acctype, Dtype>::to(outputAcc);
-    if (size_average && *total_weight > 0) {
+    if (size_average && *total_weight >= 0) {
```

Comment:
Before this PR, the following test case passes:
```
import torch
import torch.nn.functional as F

input = torch.ones(2, 3)
target = torch.zeros(2, dtype=torch.long)
weight = torch.zeros(3)
assert F.nll_loss(input, target, weight).item() == 0
assert F.nll_loss(input.cuda(), target.cuda(), weight.cuda()).item() == 0
```

After this PR, the above test case fails. In particular, `F.nll_loss(input.cuda(), target.cuda(), weight.cuda())` is now equal to NaN.

This is a pretty weird edge case (that we don't have a test for), but we should not change the behavior. Let's (1) add a test case for this somewhere and (2) fix it

Reply:
addressed